_source/_posts/2022-05-02-angular-microfrontend-auth.md

```diff
@@ -0,0 +1,443 @@
+
+**Prerequisites**
+
+- [Node](https://nodejs.org/) This project was developed using Node v16.14 with npm v8.5 
+- [Angular CLI](https://angular.io/cli)
+- [Okta CLI](https://cli.okta.com)
+
+**Table of Contents**{: .hide }
+* Table of Contents
+{:toc}
+
+
+## Angular app with micro-frontend starter using Webpack and Module Federation
+
+There's a lot in this web app! We'll use a starter code to make sure we focus on the micro-frontend-specific code. If you're dismayed that you're using a starter and not starting from scratch, don't worry. I'll provide the Angular CLI commands to recreate the structure of this starter app on the repository's README.md so you have all the instructions.
+
+Clone the [Angular Micro-frontends Example](https://github.com/oktadev/okta-angular-microfrontend-example) GitHub repo, open a terminal to the repo's filesystem location, and install the dependencies by running `npm ci`. Open the repo in your favorite IDE.
+
+Let's dive into the code! 🎉
+
+We have an Angular project with two applications and one library inside the `src/projects` directory. The two applications are named `shell` and `mfe-basket`, and the library is named `shared`. The `shell` application is the micro-frontend host, and the `mfe-basket` is a micro-frontend remote application. The library, `shared`, contains code and application state we want to share across the site. Apply the same sort of diagram shown above for this app looks like this:
+
+{% img blog/angular-microfrontend-auth/sample-app-diagram.jpg alt:"Image of the sample cupcake e-commerce site showing a breakdown of the views. The entire site is wrapped in a host named 'shell' and a micro-frontends for the 'basket' link" width:"800" %}{: .center-image }
+
+In this project, we use the `@angular-architects/module-federation` dependency to help encapsulate some of the intricacies of configuring Webpack and the Module Federation plugin. The `shell` and `mfe-basket` application have their own separate `webpack.config.js`. Open the `projects/shell/webpack.config.js` file for either the `shell` or `mfe-basket` application to see the overall structure. This file is where we add in the wiring for the hosts, remotes, shared code, and shared dependencies in the Module Federation plugin. The structure will be different if you aren't using the `@angular-architects/module-federation` dependency, but the basic idea for configuration remains the same.
+
+Let's explore sections of this config file more. 
+
+```js
+// ...imports here
+
+const sharedMappings = new mf.SharedMappings();
+sharedMappings.register(
+  path.join(__dirname, '../../tsconfig.json'),
+  [
+    '@shared'
+  ]);
+
+module.exports = {
+  // ...other very important config properties
+  plugins: [
+    new ModuleFederationPlugin({
+        library: { type: "module" },
```

Comment:
This looks to be indented 4 spaces instead of two. It might be intentional too. Feel free to ignore.